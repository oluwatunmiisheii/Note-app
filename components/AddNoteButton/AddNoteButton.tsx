import { FontAwesome } from "@expo/vector-icons";
import { View, StyleSheet } from "react-native";
import { Button } from "../Common/Button/Button";

interface AddNoteButtonProps {
  onAddButtonPress: () => void;
}

export const AddNoteButton = ({ onAddButtonPress }: AddNoteButtonProps) => {
  return (
    <View style={styles.container}>
      <Button
        onPress={onAddButtonPress}
        customStyles={styles.addButton}
        circle
        testID="add-note-button"
      >
        <FontAwesome name="plus" size={30} color="white" />
      </Button>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: "100%",
    alignItems: "center",
    marginBottom: 10,
  },
  addButton: {
    position: "absolute",
    right: 10,
    bottom: 0,
  },
});
