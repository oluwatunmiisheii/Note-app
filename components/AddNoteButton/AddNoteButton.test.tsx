import renderer from "react-test-renderer";
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import { AddNoteButton } from "./AddNoteButton";

const defaultProps = {
  onAddButtonPress: jest.fn(),
};

describe(`${AddNoteButton}`, () => {
  it("renders correctly", () => {
    const tree = renderer.create(<AddNoteButton {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onAddButtonPress when button is pressed", () => {
    render(<AddNoteButton {...defaultProps} />);
    const button = screen.getByTestId("add-note-button");
    fireEvent(button, "onPress");
    expect(defaultProps.onAddButtonPress).toHaveBeenCalled();
  });
});
