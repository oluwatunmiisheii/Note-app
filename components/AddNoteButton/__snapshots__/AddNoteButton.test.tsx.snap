// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function AddNoteButton(_ref) {
  var onAddButtonPress = _ref.onAddButtonPress;
  return (0, _jsxRuntime.jsx)(_reactNative.View, {
    style: styles.container,
    children: (0, _jsxRuntime.jsx)(_Button.Button, {
      onPress: onAddButtonPress,
      customStyles: styles.addButton,
      circle: true,
      testID: "add-note-button",
      children: (0, _jsxRuntime.jsx)(_vectorIcons.FontAwesome, {
        name: "plus",
        size: 30,
        color: "white"
      })
    })
  });
} renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 10,
      "width": "100%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "padding": 5,
            "width": "auto",
          },
          Object {
            "backgroundColor": "#007bff",
          },
        ],
        Object {
          "height": 60,
          "width": 60,
        },
        Object {
          "bottom": 0,
          "position": "absolute",
          "right": 10,
        },
      ]
    }
    testID="add-note-button"
  >
    <FontAwesomeIcon
      color="white"
      name="plus"
      size={30}
    />
  </View>
</View>
`;
